### PR TITLE
Drop currentDir where possible

### PR DESCRIPTION
**Pull request by jelly**
This drops currentDir where possible except in the navigatorCardBody, it can be dropped there after we merge https://github.com/cockpit-project/cockpit-navigator/pull/254

An alternative approach is maybe to add `src/utils.ts` and then add a helper called `getCurrentDir(path: []string) => string` for looking up the directory name.

### CODE REVIEW

**Review by martinpitt — DISMISSED** [review]
Yay linters/typecheckers? :grin:

File: src/sidebar.jsx [hunk 1]
```diff
@@ -315,7 +313,7 @@ const DropdownWithKebab = ({
             id: "edit-permissions",
             onClick: () => {
                 editPermissions(Dialogs, {
-                    selected: selected[0] || currentDirectory,
+                    selected: selected[0] || directory_name,
```
Comment (cockpituous) [inline src/sidebar.jsx:316]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-256-20240207-173252-9bf6d491-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/sidebar.jsx [hunk 2]
```diff
@@ -315,7 +313,7 @@ const DropdownWithKebab = ({
             id: "edit-permissions",
             onClick: () => {
                 editPermissions(Dialogs, {
-                    selected: selected[0] || currentDirectory,
+                    selected: selected[0] || { name: directory_name, type: "dir" },
```
Comment (jelly) [inline src/sidebar.jsx:316]: this is still utter crap, but let's imagine a way to do that via UseContext. Gotta sit down and think about that, can we use `rootInfo`?

Comment (cockpituous) [inline src/sidebar.jsx:316]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-256-20240208-090025-6814ca30-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/fileActions.jsx [hunk 3]
```diff
@@ -124,7 +124,7 @@ export const ConfirmDeletionDialog = ({
             modalTitle = cockpit.format(_("Delete file $0?"), selectedItem.name);
         } else if (selectedItem.type === "lnk") {
             modalTitle = cockpit.format(_("Delete link $0?"), selectedItem.name);
-        } else if (selectedItem.type === "dir" || selectedItem.items_cnt) {
+        } else if (selectedItem.type === "dir") {
```
Comment (cockpituous) [inline src/fileActions.jsx:127]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-256-20240212-120023-29378c53-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/fileActions.jsx [hunk 4]
```diff
@@ -169,7 +169,7 @@ export const ForceDeleteModal = ({ selected, path, initialError }) => {
             modalTitle = cockpit.format(_("Force delete file $0?"), selectedItem.name);
         } else if (selectedItem.type === "lnk") {
             modalTitle = cockpit.format(_("Force delete link $0?"), selectedItem.name);
-        } else if (selectedItem.type === "dir" || selectedItem.items_cnt) {
+        } else if (selectedItem.type === "dir") {
```
Comment (cockpituous) [inline src/fileActions.jsx:172]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-256-20240212-120023-29378c53-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/fileActions.jsx [hunk 5]
```diff
@@ -272,7 +272,7 @@ export const RenameItemModal = ({ path, selected, setHistory, setHistoryIndex })
         title = cockpit.format(_("Rename file $0"), selected.name);
     } else if (selected.type === "lnk") {
         title = cockpit.format(_("Rename link $0"), selected.name);
-    } else if (selected.type === "dir" || selected.items_cnt) {
+    } else if (selected.type === "dir") {
```
Comment (cockpituous) [inline src/fileActions.jsx:275]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-256-20240212-120023-29378c53-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/sidebar.jsx (hunk 2 above)
Comment (cockpituous) [inline src/sidebar.jsx:316]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-256-20240212-120023-29378c53-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Comment by martinpitt** [comment]
@jelly This is conflicty now, from the other PR.

File: src/fileActions.jsx (hunk 3 above)
Comment (cockpituous) [inline src/fileActions.jsx:127]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-256-20240212-150910-02953828-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/fileActions.jsx (hunk 4 above)
Comment (cockpituous) [inline src/fileActions.jsx:172]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-256-20240212-150910-02953828-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/fileActions.jsx (hunk 5 above)
Comment (cockpituous) [inline src/fileActions.jsx:275]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-256-20240212-150910-02953828-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/sidebar.jsx [hunk 6]
```diff
@@ -276,7 +275,7 @@ const DropdownWithKebab = ({
...
```
Comment (cockpituous) [inline src/sidebar.jsx:278]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-256-20240212-150910-02953828-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Pull request merged**